mtags/src/main/scala-2/scala/meta/internal/pc/InferredMethodProvider.scala

```diff
@@ -0,0 +1,496 @@
+            // we get the position of the container
+            // class because we want to add the method
+            // definition there
+            val containerClass =
+              context.lookupSymbol(classSymbol.name, _ => true)
+            // this gives me the position of the class
+            typedTreeAt(containerClass.symbol.pos) match {
+              // class case
+              case ClassDef(
+                    _,
+                    _,
+                    _,
+                    template
+                  ) =>
+                makeClassMethodEdits(template)
+              // trait case
+              case ModuleDef(
+                    _,
+                    _,
+                    template
+                  ) =>
+                makeClassMethodEdits(template)
+              case _ =>
+                // object Y {}
+                // Y.nonExistent(1,2,3)
+                typedTreeAt(containerSymbol.symbol.pos) match {
+                  // class case
+                  case ClassDef(
+                        _,
+                        _,
+                        _,
+                        template
+                      ) =>
+                    makeClassMethodEdits(template)
+                  // trait case
+                  case ModuleDef(
+                        _,
+                        _,
+                        template
+                      ) =>
+                    makeClassMethodEdits(template)
+                  case _ =>
+                    Nil
+                }
+            }
+          case _ =>
+            Nil
+        }
+      } else
+        Nil
+    }
+    val uriTextEditPairs = typedTree match {
+      case errorMethod: Ident if errorMethod.isErroneous =>
+        lastVisitedParentTrees match {
+          /**
+           * Works for apply with unknown name:
+           * ```scala
+           * method(nonExistent(param))
+           * ```
+           */
+          case _ :: (nonExistent: Apply) :: Apply(
+                Ident(containingMethod),
+                arguments
+              ) :: _ if nonExistent.isErrorTyped =>
+            makeEditsForApplyWithUnknownName(
+              arguments,
+              containingMethod,
+              errorMethod,
+              nonExistent
+            )
+          // nonExistent(param1, param2)
+          // val a: Int = nonExistent(param1, param2)
+          case (_: Ident) :: Apply(
+                containing @ Ident(
+                  nonExistent
+                ),
+                arguments
+              ) :: _ if containing.isErrorTyped =>
+            signature(
+              name = nonExistent.toString(),
+              paramsString = argumentsString(arguments).getOrElse(""),
+              retType = None,
+              postProcess = identity,
+              position = None
+            )
+          // List(1, 2, 3).map(nonExistent)
+          // List((1, 2)).map{case (x,y) => otherMethod(x,y)}
```

Comment:
Is this actually the snippet this handles? Looks like this is a simple apply `containingMethod(arguments)` The same as the one above

I don't see any check for being inside another method.

Reply:
yes, this is handling both List cases: both the non argument case, and the unpacking `case`